src/bos/server/redis_db_utils.py

```diff
@@ -110,9 +109,11 @@ def iter_values(self):
         """
         Iterate through every item in the database. Parse each item as JSON and yield it.
         """
-        for next_keys in batched(self.client.scan_iter(), 500):
-            for datastr in self.client.mget(next_keys):
+        all_keys = list(self.client.scan_iter())
+        while all_keys:
```

Comment:
ChatGPT had some suggestions, but I think what you have works well enough.

import json
import itertools

def fetch_in_batches(self, all_keys, batch_size=500):
    # Use itertools.islice for efficient batching
    while all_keys:
        # Take a slice of the first `batch_size` keys
        batch = list(itertools.islice(all_keys, 0, batch_size))
        
        # Fetch the data for this batch
        results = self.client.mget(batch)
        
        for datastr in results:
            if datastr:
                try:
                    yield json.loads(datastr)
                except json.JSONDecodeError:
                    yield None  # Or handle the error more gracefully
            else:
                yield None

        # Update `all_keys` to remove the processed ones
        all_keys = all_keys[batch_size:]

Reply:
It's hard for me to see how that would be faster, but I'm gonna test and see. Probably not for this PR, but if that IS faster, there is a mystery afoot that I want to understand.

Reply:
In my testing, the ChatGPT suggestion is slower. Both because I think the islice call is slower than just direct slicing, and because of the way it uses the try/except blocks. Plus, the try/except stuff doesn't fit with the way the rest of our db utility methods handle that. So I think for now humans still beat AI. :)